tests/test_xarrayfits.py

```diff
@@ -162,7 +162,7 @@ def test_beam_creation(beam_cube):
     cmp_data = cmp_data.reshape(xds.hdu0.shape)
     assert_array_equal(xds.hdu0.data, cmp_data)
     assert xds.hdu0.data.shape == (257, 257, 32)
-    assert xds.hdu0.dims == ("hdu0-0", "hdu0-1", "hdu0-2")
+    assert xds.hdu0.dims == ("X0", "Y0", "FREQ0")
```

Comment:
Here, the dimension names come out have the form `{CTYPE}{hdu_index}`.

I don't like the `hdu_index` all that much but this a compromise in the case where a FITS file has multiple HDUs, with possibly  different CNAME and CTYPE headers.

@paoloserra might also have some good suggestions on conventions to use here.

Reply:
The string representation looks as follows:

I've assigned the dimension names to the coordinates.

```
<xarray.Dataset> Size: 17MB
Dimensions:  (X0: 257, Y0: 257, FREQ0: 32)
Coordinates:
  * X0       (X0) float64 2kB -1.407 -1.396 -1.385 -1.374 ... 1.407 1.418 1.43
  * Y0       (Y0) float64 2kB -1.418 -1.407 -1.396 -1.385 ... 1.396 1.407 1.418
  * FREQ0    (FREQ0) float64 256B 8.56e+08 8.828e+08 ... 1.658e+09 1.685e+09
Data variables:
    beam0    (X0, Y0, FREQ0) float64 17MB dask.array<chunksize=(257, 257, 32), meta=np.ndarray>
```

Reply:
Not sure about embedding `hdu_index` in the coordinate names here. Should a fits file with multiple HDUs not be considered as separate datasets rather?

Reply:
> The alternative might be to concatenate the files into a single dataset along, say, a file dimension. Then we could have a list of HDU datasets. I'm not sure if this is more intuitive, but I'm interested in what FITS users think.

It would be up to the user to use xarray to rename the `file` dimension. I don't think a call to 

- https://docs.xarray.dev/en/latest/generated/xarray.Dataset.rename_dims.html

is onerous.

Reply:
I also dislike the embedded HDU index. And I've never even seen a multiple-HDU FITS file in the flesh. So this is inflicting a constant (albeit minor) annoyance upon ourselves for the sake of supporting a hypothetical use case which may never actually arise. 

How about a compromise: no embedding for HDU 0. Use the variable and dim names as is. But do embed the index for HDUs 1 and onwards. That way we remain annoyance-free in real life, and are still compatible with the hypotheticals.

For top prize, add an ``hdu`` parameter to ``xarray_from_fits()`` specifying which HDU(s) to read. Default is all. And if you use this to pull out a single HDU (that is not 0), also do not embed the index.

I don't think we need a file dimension -- the current multiple dataset paradigm seems just fine to me.

Reply:
Resolving this, but I'll highlight the new regime in the new changes that make this obsolete.